common/cluster/metadata.go

```diff
@@ -409,13 +395,8 @@ func (m *metadataImpl) RegisterMetadataChangeCallback(callbackId any, cb Callbac
 	m.clusterLock.RLock()
 	for clusterName, clusterInfo := range m.clusterInfo {
 		oldEntries[clusterName] = nil
-		newEntries[clusterName] = &ClusterInformation{
-			Enabled:                clusterInfo.Enabled,
-			InitialFailoverVersion: clusterInfo.InitialFailoverVersion,
-			RPCAddress:             clusterInfo.RPCAddress,
-			ShardCount:             clusterInfo.ShardCount,
-			version:                clusterInfo.version,
-		}
+		newEntries[clusterName] = ShallowCopyClusterInformation(&clusterInfo)
+		newEntries[clusterName].Tags = nil
```

Comment:
I'm not sure why we do this actually. @yux0 , why do we not copy over the `Tags` field here?

Reply:
Nevermind, talked offline and seems like this wasn't necessary. Now doing a regular full copy